app/services/user_opportunity_discovery.py

```
@@ -1890,49 +2092,175 @@ async def _scan_spot_breakout_opportunities(
+                            )
+                            breakout_probability = (
+                                (breakout_probability or 0.0) / 100.0
+                                if breakout_probability is not None
+                                else 0.0
+                            )
+
+                        breakout_probability = max(0.0, min(breakout_probability or 0.0, 1.0))
+
+                        # Track ALL signals for transparency
+                        self.logger.info(
+                            f"🎯 BREAKOUT SIGNAL ANALYSIS",
```

Ruff (0.14.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)